hashes/src/sha256.rs

```diff
@@ -433,22 +449,6 @@ impl Midstate {
 }
 
 impl HashEngine {
```

Comment:
I've looked at that more than once over the years, I believe its because the `process()` function is so low level and long that its better having it down the bottom. Obviously I didn't write it though. @apoelstra ?